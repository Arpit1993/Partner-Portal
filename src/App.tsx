/* eslint-disable react/jsx-no-constructed-context-values */
import './App.css';
import { Route, Routes, useNavigate, Navigate } from 'react-router-dom';
import { useEffect } from 'react';
import ErrorBoundary from 'components/ErrorBoundary/ErrorBoundary';
import ForgotPassword from 'pages/ForgotPassword/ForgotPassword';
import NotFoundNew from 'pages/NotFound/NotFound';
import LoginNew from 'pages/LoginScreen/LoginScreen';
import FirwareSDKDocumentation from 'pages/FirmwareAndSDK/FirmwareAndSDK';
import { isAuthenticated, logout } from 'utils/';
import UserManagementNew from 'pages/UserManagement/UserManagement';
import AddUser from 'components/AddUser/AddUser';
import DeviceManagement from 'pages/DeviceManagement/DeviceManagement';
import DeviceManageFirmwareAndDID from 'pages/DeviceFirmareAndDID/DeviceFirmareAndDID';
// import ProdDeviceManagement from 'pages/ProdDeviceManagement/ProdDeviceManagement';
// import ProdDeviceManageFirmwareAndDID from 'pages/ProdDeviceFirmware/ProdDeviceFirmware';
import AppVersion from 'pages/AppVersion/AppVersion';
import UpdateDID from 'components/UpdateDID/UpdateDID';
import UploadDatasheet from 'components/UploadDatasheet/UploadDatasheet';
import ResetPassword from 'pages/ResetPassword/ResetPassword';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import ImageCarousel from './components/ImageCarousel/ImageCarousel';

function App() {
  const navigate = useNavigate();

  useEffect(() => {
    const currentPath = window.location.pathname;

    // Redirect only if user is not authenticated and not on the login/forgot-password page
    if (
      !isAuthenticated() &&
      !['/login', '/forgotPassword', '/resetPassword'].includes(currentPath)
    ) {
      navigate('/login');
      logout();
    }
  }, [navigate]);

  return (
    <div className="App" style={{ height: '100vh' }}>
      <ErrorBoundary>
        <Routes>
          <Route path="/login" Component={LoginNew} />
          <Route path="forgotPassword" Component={ForgotPassword} />
          <Route path="resetPassword" Component={ResetPassword} />

          <Route element={<PrivateRoute />}>
            <Route path="/engineering">
              <Route path="device-management" Component={DeviceManagement} />
              <Route
                path="firmware/:deviceId"
                Component={DeviceManageFirmwareAndDID}
              />
            </Route>

            <Route path="/production">
              <Route path="device-management" Component={DeviceManagement} />

              <Route
                path="firmware/:deviceId"
                Component={DeviceManageFirmwareAndDID}
              />
            </Route>
            <Route
              path="/firmware-management"
              Component={FirwareSDKDocumentation}
            />
            <Route path="/user-management" Component={UserManagementNew} />
            <Route path="/user-management/add-user" Component={AddUser} />
            <Route path="/app-version" Component={AppVersion} />
            <Route path="/image-carousel" Component={ImageCarousel} />
            <Route path="/upload-datasheet" Component={UploadDatasheet} />
            <Route path="force-ota-update" Component={UpdateDID} />
          </Route>
          <Route path="/not-found-new" Component={NotFoundNew} />
          <Route
            path="/"
            element={<Navigate to="/engineering/device-management" />}
          />
          <Route path="*" Component={NotFoundNew} />
        </Routes>
      </ErrorBoundary>
    </div>
  );
}

export default App;
