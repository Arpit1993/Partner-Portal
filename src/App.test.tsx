import React from 'react';
import { render } from '@testing-library/react';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import App from 'App';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const renderWithRouter = (route = '/') => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  window.history.pushState({}, 'Test page', route);
  return {
    ...render(
      <Provider store={store}>
        <RouterProvider
          router={createBrowserRouter([{ path: '*', element: <App /> }])}
        />
      </Provider>
    )
  };
};

test('renders app page', () => {
  const screen = renderWithRouter();
  expect(screen.baseElement).toMatchSnapshot();
});
