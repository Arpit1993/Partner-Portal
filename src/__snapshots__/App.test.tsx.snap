// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app page 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div
        class="login-container"
      >
        <div
          class="top"
        >
          <h1
            class="hidden"
            id="title"
          >
            <span
              id="logo"
            >
              Instaview.
              <span>
                AI
              </span>
            </span>
          </h1>
        </div>
        <div
          class="login-box animated fadeInUp"
        >
          <div
            class="box-header"
          >
            <h2>
              Log In
            </h2>
          </div>
          <span>
            Username
          </span>
          <br />
          <input
            id="username"
            type="text"
            value=""
          />
          <br />
          <span>
            Password
          </span>
          <br />
          <input
            id="password"
            type="password"
            value=""
          />
          <br />
          <button
            class="sign-in"
            disabled=""
            type="button"
          >
            Sign In
          </button>
          <br />
          <div
            class="separate"
          >
            Or
          </div>
          <div
            class="social-login"
          >
            <button
              class="google-login"
              type="button"
            >
              <img
                alt="google"
                src="google-icon.png"
              />
              Sign in with Google
            </button>
            <button
              class="apple-login"
              type="button"
            >
              <img
                alt="google"
                src="apple-icon.png"
              />
              Sign in with Apple
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
